opm/simulators/flow/KeywordValidation.hpp

```diff
@@ -0,0 +1,50 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_KEYWORDVALIDATION_HEADER_INCLUDED
+#define OPM_KEYWORDVALIDATION_HEADER_INCLUDED
+
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace Opm
+{
+
+class ErrorGuard;
+class ParseContext;
+
+namespace KeywordValidation
+{
+
+    struct UnsupportedKeywordProperties {
+        bool critical;
+        std::optional<std::string> message;
+    };
```

Comment:
I guess this also needs some datastructures geared towards unsupported items; for an unsupported/partially supported item I guess you can choose between an allowed_values_list, illegal_values_list, only_default - or others. This might get quite low level (and ugly ...) in terms of opm-common/Deck fiddling.

Reply:
Yes, but I hope we can keep it simple at this point in time. The current implementation (before this PR) had to handle each keyword item manually in the code, and only had a binary approach (one value supported, one value not supported). This is already miles further.